test/Dialect/FIRRTL/layers-errors.mlir

```diff
@@ -0,0 +1,92 @@
+// RUN: circt-opt %s -split-input-file -verify-diagnostics
+
+firrtl.circuit "CastAwayColor" {
+  firrtl.layer @A bind {}
+  firrtl.module @CastAwayColor() {
+    %w = firrtl.wire : !firrtl.probe<uint<1>, @A>
+    // expected-error @below {{cannot discard layer requirements of input ref}}
+    // expected-note  @below {{discarding layer requirements: @A}}
+    %s = firrtl.ref.cast %w : (!firrtl.probe<uint<1>, @A>) -> !firrtl.probe<uint<1>>
+  }
+}
+
+// -----
+
+firrtl.circuit "Top" {
+  firrtl.layer @A bind {}
+  firrtl.module @UncoloredDefineUnderLayerblock(out %o : !firrtl.probe<uint<1>>) {
+    %c = firrtl.constant 0 : !firrtl.uint<1>
+    %r = firrtl.ref.send %c : !firrtl.uint<1>
+    firrtl.layerblock @A {
+      // expected-error @below {{'firrtl.ref.define' op has more layer requirements than destination}}
+      // expected-note  @below {{additional layers required: @A}}
+      firrtl.ref.define %o, %r : !firrtl.probe<uint<1>>
+    }
+  }
+}
+
+// -----
+
+firrtl.circuit "Top" {
+  firrtl.layer @A bind {}
+  firrtl.module @Top(out %o : !firrtl.probe<uint<1>>) {
+    firrtl.layerblock @A {
+      %c = firrtl.constant 0  : !firrtl.uint<1>
+      %r = firrtl.ref.send %c : !firrtl.uint<1>
+      // expected-error @below {{'firrtl.ref.define' op has more layer requirements than destination}}
+      // expected-note  @below {{additional layers required: @A}}
+      firrtl.ref.define %o, %r : !firrtl.probe<uint<1>>
+    }
+  }
+}
```

Comment:
Is this test checking anything given that the verification is only happening at the `firrtl.ref.define` and the `firrtl.ref.send` doesn't factor into the check? I may be missing something.

Reply:
This test is really here to assert that the location of the src ref doesn't matter.